server/app/assets/javascripts/sort_selector.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * Responsible for controlling the sorting mechanism used by QuestionBankController. Elements (tagged with elementSelector class)
+ * have data attributes with the value to be compared and will sort list using the value. The data attribute is defined
+ * in QuestionSortOption.
+ *
+ * E.g. Sort by Name A-Z will look up the value for data attribute tiname-asc and sort the list by that value.
+ *
+ * @param listSelector selector for finding the portion of the document that contains the sublistSelector and elementSelector
+ * @param sublistSelector selector for the parent element of all of the elements to be sorted
+ * @param elementSelector individual elements to be sorted
+ */
+export function sortSelectorElements(
```

Comment:
What does "selector" mean here in the method name? If it's for sorting QuestionBanks (per the description) then would sortQuestionBank be clearer? Otherwise it sounds like it's a generic method for sorting... something :)

Reply:
I originally named it sortElements (which is still generic sounding!) and then renamed it to be sortSelectorElements (still too generic!) and I think maybe a better name is sortElementsByDataAttributes?

Reply:
I plan on using this for the TrustedIntermediaryController as well so didn't want to use a name that was too narrow in scope. 

https://github.com/civiform/civiform/pull/7600/files (branch  with my changes to add TrustedIntemediaryController)

Reply:
Makes sense - I do like sortElementsByDataAttributes better, since it captures the magic within the method.

Only remaining question is then shouldn't the jsdoc reflect the generic nature of the method and not refer to QuestionBank specifically?

Reply:
Was trying too hard to write a method for the future without referencing non-existent methods and planned methods which worked about as okay as planned haha I should have been more explicit and communicative in the PR description. I'm going to update this file with the TrustedIntermediaryController reference in the next PR that adds that file.